lib/common/widgets/bottom_nav_bar.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../../data/repository/local_storage.dart';
+import '../../di/injectable.dart';
+import '../utils/constants/pref_constants.dart';
+
+class CustomBottomNavigationBar extends StatefulWidget {
+  const CustomBottomNavigationBar(
+      {super.key, required this.selectedPageIndex, required this.onPageChange});
+
+  final int selectedPageIndex;
+  final Function(int) onPageChange;
+
+  @override
+  State<CustomBottomNavigationBar> createState() =>
+      _CustomBottomNavigationBarState();
+}
+
+class _CustomBottomNavigationBarState extends State<CustomBottomNavigationBar> {
+  /// This is used for the swipe drag gesture on the bottom nav bar
+  var localStorage = getIt<LocalStorage>();
+  bool bottomNavBarSwipeGestures = false, bottomNavBarDoubleTapGestures = false;
+
+  double _dragStartX = 0.0;
+  double _dragEndX = 0.0;
+
+  // Handles drag on bottom nav bar to open the drawer
+  void _handleDragStart(DragStartDetails details) {
+    _dragStartX = details.globalPosition.dx;
+  }
+
+  // Handles drag on bottom nav bar to open the drawer
+  void _handleDragUpdate(DragUpdateDetails details) async {
+    _dragEndX = details.globalPosition.dx;
+  }
+
+  // Handles drag on bottom nav bar to open the drawer
+  void _handleDragEnd(DragEndDetails details, BuildContext context) async {
+    if (widget.selectedPageIndex != 0) return;
+
+    if (bottomNavBarSwipeGestures == false) return;
+
+    double delta = _dragEndX - _dragStartX;
+
+    // Set some threshold to also allow for swipe up to reveal FAB
+    if (delta > 20) {
+      if (context.mounted) Scaffold.of(context).openDrawer();
```

Comment:
Since this is a synchronous method, we don't really need to check for mounted. It's unlikely for the method to be called after widget is unmounted

Reply:
Resolved this too